hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -132,14 +179,19 @@ enum Term extends Statement:
    * The symbol representing the evaluation result of the term. This
    * symbol is resolved during the resolution stage.
    */
-  def resolvedSymbol: Opt[Symbol] = this match
-    case ref: Ref => ref.resSym
-    case sel: Sel => sel.sym
-    case sel: SynthSel => sel.sym
-    case sel: SelProj => sel.sym
-    case app: App => app.sym
-    case tyApp: TyApp => tyApp.sym
-    case _ => N
+  def resolvedSymbol: Opt[Symbol] =
+    // TODO: encode mutable symbols into expansions
+    this match
+      case r: Resolvable if r.hasExpansion => r.instantiate
+      case t => t
+    match
+      case ref: Ref => ref.resSym
+      case sel: Sel => sel.sym
+      case sel: SynthSel => sel.sym
+      case sel: SelProj => sel.sym
+      case app: App => app.sym
+      case tyApp: TyApp => tyApp.sym
+      case _ => N
```

Comment:
I think it is a little ugly here as there are now multiple source of truth (of the resolved symbol). But since there are enough changes in this PR, could we merge it before refactoring this place?